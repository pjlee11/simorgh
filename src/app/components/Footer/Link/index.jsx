import React from 'react';
import styled from 'styled-components';
import { string, bool } from 'prop-types';
import { C_WHITE, GEL_SPACING } from '../../../lib/constants/styles';

const StyledLink = styled.a`
  padding: ${GEL_SPACING} 0 ${GEL_SPACING};
  color: ${C_WHITE};
  font-weight: 700; /* Used instead of Reith Sans Bold since it is not worth the performance cost in this case. */
  text-decoration: none;
  display: ${({ inline }) => (inline ? 'inline' : 'block')};
`;

const StyleLinkText = styled.span`
  ${StyledLink}:hover &,
  ${StyledLink}:focus & {
    padding-bottom: 2px;
    border-bottom: 2px solid ${C_WHITE};
  }
`;

const Link = ({ text, href, inline }) => (
  <StyledLink inline={inline} href={href}>
    <StyleLinkText>{text}</StyleLinkText>
  </StyledLink>
);

Link.defaultProps = {
  inline: false,
};

Link.propTypes = {
  href: string.isRequired,
  text: string.isRequired,
  inline: bool,
};

export default Link;
