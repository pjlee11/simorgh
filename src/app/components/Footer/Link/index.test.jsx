import React from 'react';
import { shouldMatchSnapshot } from '../../../helpers/tests/testHelpers';
import Link from './index';

const text = 'Some link text';
const href = 'https://www.bbc.com/';

describe('Link', () => {
  shouldMatchSnapshot(
    'should render a standard block link',
    <Link text={text} href={href} />,
  );
  shouldMatchSnapshot(
    'should render an inline link after passing an inline prop',
    <Link text={text} href={href} inline />,
  );
});
