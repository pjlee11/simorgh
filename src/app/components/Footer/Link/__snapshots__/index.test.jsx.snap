// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render a standard block link 1`] = `
.c1 {
  padding: 0.5rem 0 0.5rem;
  color: #FFFFFF;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
}

.c0:hover .c2,
.c0:focus .c2 {
  padding-bottom: 2px;
  border-bottom: 2px solid #FFFFFF;
}

<a
  className="c0 c1"
  href="https://www.bbc.com/"
>
  <span
    className="c2"
  >
    Some link text
  </span>
</a>
`;

exports[`Link should render an inline link after passing an inline prop 1`] = `
.c1 {
  padding: 0.5rem 0 0.5rem;
  color: #FFFFFF;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline;
}

.c0:hover .c2,
.c0:focus .c2 {
  padding-bottom: 2px;
  border-bottom: 2px solid #FFFFFF;
}

<a
  className="c0 c1"
  href="https://www.bbc.com/"
>
  <span
    className="c2"
  >
    Some link text
  </span>
</a>
`;
